openedx/tests/xblock_integration/test_review_xblock.py

```diff
@@ -0,0 +1,510 @@
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section3_actual.location.name,
+            }
+        ))
+
+        user = User.objects.get(email=self.STUDENTS[0]['email'])
+        crum.set_current_user(user)
+        result_urls = get_review_ids.get_problems(num_desired, self.course_actual.id)
+
+        url_beginning = 'https://courses.edx.org/xblock/block-v1:DillonX/DAD101xr/3T2017+type@problem+block@'
+        expected_urls = [
+            (url_beginning + 'Problem_1', True, 0),
+            (url_beginning + 'Problem_2', True, 0),
+            (url_beginning + 'Problem_3', True, 0),
+            (url_beginning + 'Problem_4', True, 0),
+            (url_beginning + 'Problem_5', True, 0),
+            (url_beginning + 'Problem_6', True, 0)
+        ]
+
+        # Since the problems are randomly selected, we have to check
+        # the correct number of urls are returned.
+        count = 0
+        for url in expected_urls:
+            if url in result_urls:
+                count += 1
+        self.assertEqual(count, num_desired)
+
+    @ddt.data(1, 2, 3, 4, 5)
+    def test_review_problem_urls_unique_problem(self, num_desired):
+        """
+        Verify that the URLs returned from the Review xBlock are valid and
+        correct URLs for the problems the learner has seen. This test will give
+        a unique problem to a learner and verify only that learner sees
+        it as a review
+
+        TODO: This test is hardcoded to assume the number of review
+        problems to show is 5. Ideally this should
+        be dependent on the number of desired review problems
+        """
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_actual)
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_review)
+
+        # Loading problems so the learner has enough problems in the CSM
+        self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section1_actual.location.name,
+            }
+        ))
+        self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section3_actual.location.name,
+            }
+        ))
+
+        user = User.objects.get(email=self.STUDENTS[0]['email'])
+        crum.set_current_user(user)
+        result_urls = get_review_ids.get_problems(num_desired, self.course_actual.id)
+
+        url_beginning = 'https://courses.edx.org/xblock/block-v1:DillonX/DAD101xr/3T2017+type@problem+block@'
```

Comment:
I'm not entirely happy about having this hard-coded to courses.edx.org, but I think it has to be fixed in the xblock's code and not here.